### ISSUE TITLE
CreateIndex with Attachment will fail

### ISSUE THREAD

**Issue by kyakhta** [elastic/elasticsearch-net#1826]
```c#
 client.CreateIndex(Index, c => c
                .Mappings(m => m.Map<Docx>(p => p
                .Properties(ps => ps.Attachment(a => a.FileField(t => t.Name(e => e.Foo))))
                    .Properties(ps => ps.String(s => s.Name(o => o.Title))))));
```

will fail on `a => a.FileField(t => t.Name(e => e.Foo)` because `AttachmentPropertyDescriptor.Dictionary` is not initialized and has null value:

```c#
private AttachmentPropertyDescriptor<T> SetMetadataField<TDescriptor, TInterface>(Func<TDescriptor, TInterface> selector, string fieldName)
			where TDescriptor : TInterface, new()
			where TInterface : IProperty
		{
			selector.ThrowIfNull(nameof(selector));
			var type = selector(new TDescriptor());
			**this.Dictionary[fieldName] = type;**
			return this;
		}
```

**Comment by russcam** [comment]
Thanks for reporting @kyakhta :+1:

**Comment by gmarz** [comment]
Closed via #1827
